pages/publicfiles/PublicFilesHandler.inc.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * @file pages/publicfiles/PublicFilesHandler.inc.php
+ *
+ * Copyright (c) 2013-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PublicFilesHandler
+ * @ingroup pages_publicfiles
+ *
+ * @brief Handle requests for public files functions.
+ */
+
+import('classes.handler.Handler');
+import('lib.pkp.classes.file.FileManager');
+
+class PublicFilesHandler extends Handler {
+	
+	/* @var fileManager FileManager object (see lib/pkp/classes/file/FileManager.inc.php) */
+	var $fileManager;
+	
+	//Constructor
+	function PublicFilesHandler() {
+		$this->fileManager = new FileManager();
+	}
+	
+	/**
+	 * Delete a file in a user's public files directory
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return bool
+	 */
+	function delete($args, $request) {
+		$user = $request->getUser();
+		if(!$user) {
+			return false;
+		}
+		
+		$sessionManager = SessionManager::getManager();
+		$userSession = $sessionManager->getUserSession();
+		$journalId = $userSession->getSessionVar('journalId');
+		$fileName = $args[1];
+		
+		$publicDir = $this->_getRealPublicFilesDir($journalId);
+		$filePath = $publicDir . $fileName;
+
+		if($this->fileManager->deleteFile($filePath)) {
+			if (isset($_SERVER['HTTP_REFERER'])) {
+				$request->redirectUrl($_SERVER['HTTP_REFERER']);
+			}
+			return true;
+		}
+	}
+
+
+	/**
+	 * Download a file from a role's public files directory
+	 * @param $args array
+	 * @return bool
+	 */
+	function download($args) {
+		$journalId = $args[0];
```

Comment:
same from above, why you need to pass the journal id if you can retrieve the journal using the page router?